net.c

```diff
@@ -2339,6 +2306,52 @@ int _ifcfg_write(char *device, ifcfg_t *ifcfg)
   return ok;
 }
 
+// update the sysconfig file *filename* with the data in *slist*
+//
+// slist is mutated by appending "=" to the keys
+void update_sysconfig(slist_t *slist, char *filename, char *filename_tmp) {
+  slist_t *sl;
+  FILE *fp, *fp2;
+
+  log_info("adjusting network/config:\n");
+
+  // it's easier below if we append the '=' to the keys
+  for(sl = slist; sl; sl = sl->next) {
+    strprintf(&sl->key, "%s=", sl->key);
+  }
+
+  if((fp = fopen(filename, "r"))) {
+    char buf[4096];
+
+    // we allow open to fail and check fp2 for NULL later
+    fp2 = fopen(filename_tmp, "w");
+
+    while(fgets(buf, sizeof buf, fp)) {
+      if(*buf && *buf != '#' && !isspace(*buf)) {
+        for(sl = slist; sl; sl = sl->next) {
+          if(!strncmp(buf, sl->key, strlen(sl->key))) {
```

Comment:
i think that this comparison can corrupt values when a file is written in not so expected order. Lets have ifcfg file:
```
IPADDR_0=1.1.1.1
IPADDR=2.2.2.2
```
and ```slist``` with key: IPADDR and value 3.3.3.3. Then updating the file should end with
```
IPADDR_0=3.3.3.3
IPADDR=2.2.2.2
```
which is incorrect.